### PR TITLE
feat: expand and rework bandit faction

### PR DESCRIPTION
**Pull request by Uberbagel**

### CODE REVIEW

File: mod_reforged/hooks/config/spawnlist_master.nut [hunk 1]
```diff
@@ -1,5 +1,133 @@
...
+::Const.World.Spawn.Troops.BanditMarksman.Strength = 24;
+::Const.World.Spawn.Troops.BanditMarksman.Cost = 24;
+::Const.World.Spawn.Troops.BanditRaiderWolf.Strength = 30;
+::Const.World.Spawn.Troops.BanditRaiderWolf.Cost = 30;
+::Const.World.Spawn.Troops.BanditRaider.Strength = 30;
+::Const.World.Spawn.Troops.BanditRaider.Cost = 30;
 ::Const.World.Spawn.Troops.BanditLeader.Strength = 40;
-::Const.World.Spawn.Troops.BanditLeader.Cost = 35;
+::Const.World.Spawn.Troops.BanditLeader.Cost = 40;
+::Const.World.Spawn.Troops.HedgeKnight.Strength = 45;
```
Comment (LordMidas) [inline mod_reforged/hooks/config/spawnlist_master.nut:17]: You set only the Strength, but not the Cost for HedgeKnight, Swordmaster, MasterArcher. Please set the Cost too.

File: mod_reforged/hooks/entity/tactical/enemies/bandit_poacher.nut [hunk 2]
```diff
@@ -28,17 +28,61 @@
 		this.getSprite("helmet_damage").Saturation = 0.85;
 		this.getSprite("shield_icon").Saturation = 0.85;
 		this.getSprite("shield_icon").setBrightness(0.85);
-		// this.m.Skills.add(this.new("scripts/skills/actives/rotation")); // Replaced with perk
-		// this.m.Skills.add(this.new("scripts/skills/actives/recover_skill")); // Replaced with perk
+		// this.m.Skills.add(::new("scripts/skills/actives/rotation")); // Replaced with perk
```
Comment (LordMidas) [inline mod_reforged/hooks/entity/tactical/enemies/bandit_poacher.nut:31]: You say rotation is replaced with perk but there is no Rotation perk on this character.

File: scripts/!mods_preload/mod_reforged.nut [hunk 3]
```diff
@@ -2,7 +2,8 @@
 	Version = "0.1.22",
 	ID = "mod_reforged",
 	Name = "Reforged Mod",
-	GitHubURL = "https://github.com/Battle-Modders/mod-reforged"
+	GitHubURL = "https://github.com/Battle-Modders/mod-reforged",
+	ItemTable = {}
```
Comment (LordMidas) [inline scripts/!mods_preload/mod_reforged.nut:6]: This is part of the ItemTables PR, please remove from this one.

File: scripts/entity/tactical/enemies/rf_bandit_marauder.nut [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	    				::Reforged.Skills.addPerkGroupOfEquippedWeapon(this, 4);
+	    				this.m.Skills.add(this.new("scripts/skills/perks/perk_rf_exploit_opening"));
+	    				this.m.Skills.add(this.new("scripts/skills/perks/perk_rf_rebuke"));
+	    				break;
+	    			case "weapon.woodcutters_axe":
+	    				::Reforged.Skills.addPerkGroupOfEquippedWeapon(this, 4);
+	    				this.m.Skills.add(this.new("scripts/skills/perks/perk_coup_de_grace"));
+	    				this.m.Skills.add(this.new("scripts/skills/perks/perk_crippling_strikes"));
+	    				this.m.Skills.add(this.new("scripts/skills/perks/perk_sundering_strikes"));
+	    				this.m.Skills.removeByID("actives.rf_bearded_blade");
```
Comment (LordMidas) [inline scripts/entity/tactical/enemies/rf_bandit_marauder.nut:114]: This removal of Bearded Blade is not reliable. As Bearded Blade is granted by the Axe Mastery perk, if he unequips and then re-equips an axe, he will get the Bearded Blade skill. 

I see you do this in some other entities too.

I currently can't think of a clean way to permanently make Bearded Blade unavailable to an entity, but I can take a look at that once I am back.

File: scripts/entity/tactical/enemies/rf_bandit_outlaw.nut [hunk 5]
```diff
@@ -0,0 +1,139 @@
+this.rf_bandit_outlaw <- this.inherit("scripts/entity/tactical/human", {
+	m = {
+		IsLow = false
```
Comment (LordMidas) [inline scripts/entity/tactical/enemies/rf_bandit_outlaw.nut:3]: This `IsLow` can probably be removed?

**Review by LordMidas — CHANGES_REQUESTED** [review]

File: scripts/entity/tactical/enemies/rf_bandit_marauder.nut (hunk 4 above)
Comment (LordMidas) [reply scripts/entity/tactical/enemies/rf_bandit_marauder.nut:114]: I have created an issue (#154) for this and assigned it to myself. Once I am back home I can see how to do that. For now we can merge this in as is.

**Review by LordMidas — APPROVED** [review]
The commit message needs cleanup - then can be merged.

**Comment by Uberbagel** [comment]
I noticed some errors and made some changes. Please check the most recent commit.

File: mod_reforged/hooks/config/global.nut [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            case ::Const.EntityType.RF_BanditScoundrel:
+            case ::Const.EntityType.RF_BanditRobber:
+            case ::Const.EntityType.RF_BanditHunter:
+            case ::Const.EntityType.RF_BanditVandal:
+            case ::Const.EntityType.RF_BanditPillager:
+            case ::Const.EntityType.RF_BanditOutlaw:
+            case ::Const.EntityType.RF_BanditBandit:
+            case ::Const.EntityType.RF_BanditHighwayman:
+            case ::Const.EntityType.RF_BanditMaruader:
```
Comment (Darxo) [inline mod_reforged/hooks/config/global.nut:61]: typo

**Review by Darxo — CHANGES_REQUESTED** [review]

**Review by LordMidas — APPROVED** [review]

File: scripts/mods/mod_reforged/dynamic_spawns/units/bandit_units.nut [hunk 7]
```diff
@@ -0,0 +1,79 @@
+// We adjust the cost of Vanilla units already registered with the Dynamic Spawn Framework:
```
Comment (Darxo) [inline scripts/mods/mod_reforged/dynamic_spawns/units/bandit_units.nut:1]: this comment is no longer necessary. The lines it referenced have been removed

**Review by Darxo — CHANGES_REQUESTED** [review]

**Review by Darxo — APPROVED** [review]

**Pull request merged**